src/main/java/com/cloudbees/jenkins/GitHubPushCause.java

```diff
@@ -37,5 +37,19 @@ public GitHubPushCause(File pollingLog, String pusher) throws IOException {
     public String getShortDescription() {
         return format("Started by GitHub push by %s", trimToEmpty(pushedBy));
     }
+
+    @Override
+    public boolean equals(Object o) {
+        return o instanceof GitHubPushCause
+                && this.pushedBy.equals(((GitHubPushCause) o).pushedBy)
+                && super.equals(o);
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = super.hashCode();
+        hash = 89 * hash + (this.pushedBy != null ? this.pushedBy.hashCode() : 0);
```

Comment:
Could also use `java.util.Objects` for these two.

Reply:
👍